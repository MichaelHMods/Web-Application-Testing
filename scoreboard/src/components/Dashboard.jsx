import React, {Component} from "react";


class Dashboard extends Component {
    constructor() {
        super();
        this.state = {
            strike: "",
            ball: "",
            foul: "",
            hit: "",
        };
    }

    render() {
        return (
            <div className="main-container">
                <button>STRIKE</button>
                <button>BALL</button>
                <button>FOUL</button>
                <button>HIT</button>
            </div>
        );
    }
}

export default Dashboard;
