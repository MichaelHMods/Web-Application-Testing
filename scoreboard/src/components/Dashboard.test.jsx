import React from "react"
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Dashboard from './Dashboard';


afterEach(cleanup);


it("renders", () => {
    const { asFragment } = render(<Dashboard text="ball"/>); 
    expect(asFragment()).toMatchSnapshot();
});

it('inserts text in div', () => {
    const {getByTestId, getByText} = render(<Dashboard/>);
    expect(getByTestId("divtag")).toHaveTextContent('ball');
    expect(getByText('ball')).toHaveClass('fancy-div');
})


// instanceof("render", () => {
    // const { asFragment } = render(<Header text="Hello" />);
    // expect(asFragment()).toMatchSnapshot();
// });
/*
it("inserts text in h1", () => {
    const { getByTestId, getByText} = render(<Header text="Hello!" />);
});
expect(getByTestId('h1tag)).toHaveTextContent("Hello");
// expect( getbytext('Hello!')).toHaveClass('fancy-h1');
//you can pass in any text to compare it with the tohavetextcontent function.
});
on the h1 data-testid ="h1tag">
on the h1 className="fancy=h1">
*/
// yarn test
// u
//header file
/*
import of react
export default Header ({text})
return header tag h2 tag {text}
close
*/