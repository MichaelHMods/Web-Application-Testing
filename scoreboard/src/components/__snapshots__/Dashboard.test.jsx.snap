// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="fancy-div"
    data-testid="divtag"
  >
    <div
      class="sc-bdVaJa dGBYUF testDiv"
    >
      <h3>
        DASHBOARD JS COMPONENT
      </h3>
      <div
        class="fancy-div"
        data-testid="divtag"
      />
      <div
        class="sc-bxivhb dEQSLd"
      >
        <div
          class="sc-htpNat hQgnAa"
        >
          <div
            class="sc-ifAKCX ktDDNN counter"
          >
            0
          </div>
          <button
            class="testButton"
          >
            BALL
          </button>
          <p>
            Ball Count: 0 after 4 balls
          </p>
        </div>
        <div
          class="sc-htpNat hQgnAa"
        >
          <div
            class="sc-ifAKCX ktDDNN counter"
          >
            0
          </div>
          <button
            class="testButton"
          >
            STRIKE
          </button>
          <p>
            Strike Count: 0 after 3 strikes
          </p>
        </div>
        <div
          class="sc-htpNat hQgnAa"
        >
          <div
            class="sc-EHOje cfUrMf counter"
          >
            <button
              class="testButton"
            >
              FOUL
            </button>
          </div>
          <p>
            Foul Count: up to 2 fouls, also increases strike count?
          </p>
        </div>
        <div
          class="sc-htpNat hQgnAa"
        >
          <div
            class="sc-EHOje cfUrMf counter"
          >
            <button
              class="testButton"
            >
              HIT
            </button>
          </div>
          <p>
            Hit Count: no params defined in the ReadMe...
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
